municipal_finance/update/cash_flow_v2.py

```diff
@@ -0,0 +1,95 @@
+
+import csv
+import requests
+
+from functools import reduce
+from collections import namedtuple
+from contextlib import closing
+
+from django.db import transaction
+
+from ..models import (
+    CflowItemsV2,
+    CflowFactsV2,
+    AmountTypeV2,
+)
+
+from .utils import (
+    all_to_code_dict,
+    period_code_details,
+    group_rows,
+    delete_existing,
+)
+
+
+CashFlowRow = namedtuple(
+    "CashFlowRow",
+    (
+        "demarcation_code",
+        "period_code",
+        "item_code",
+        "amount",
+    ),
+)
+
+
+class CashFlowReader(object):
+
+    def __init__(self, data):
+        self._reader = csv.reader(data)
+
+    def __iter__(self):
+        for row in map(CashFlowRow._make, self._reader):
+            yield row
+
+
+def row_to_obj(amount_types, items, row: CashFlowRow):
+    (
+        financial_year,
+        amount_type_code,
+        period_length,
+        financial_period
+    ) = period_code_details(row.period_code)
+    amount = int(row.amount) if row.amount else None
```

Comment:
why int? I'm pretty sure floats or decimals should be allowed depending on which type we're using in the db - make sure to fix this on all the cubes

Reply:
nope! we're using bigint in the DB! interesting!